packages/react-instantsearch-core/src/lib/useSearchResults.ts

```diff
@@ -16,9 +16,13 @@ export type SearchResultsApi = {
 export function useSearchResults(): SearchResultsApi {
   const search = useInstantSearchContext();
   const searchIndex = useIndexContext();
-  const [searchResults, setSearchResults] = useState(() =>
-    getIndexSearchResults(searchIndex)
-  );
+  const [searchResults, setSearchResults] = useState(() => {
+    const indexSearchResults = getIndexSearchResults(searchIndex);
+    return {
+      results: indexSearchResults.results,
+      scopedResults: indexSearchResults.scopedResults,
+    };
+  });
```

Comment:
```suggestion
  const [searchResults, setSearchResults] = useState(() =>
    getIndexSearchResults(searchIndex)
  );
```

I don't think this diff is needed?

Reply:
It is, otherwise we would have `recommendResults` in `useSearchResults` which we haven't decided yet. Problem is `getIndexSearchResults` is both used by this and `getServerState`

Reply:
ok sounds fair, maybe a comment to explain that this is to prevent recommend results? not sure if needed though